cpp/include/raft/linalg/detail/transpose.cuh

```diff
@@ -32,6 +32,77 @@ namespace raft {
 namespace linalg {
 namespace detail {
 
+template <typename IndexType, int TILE_DIM, int BLOCK_ROWS>
+RAFT_KERNEL transpose_half_kernel(IndexType n_rows,
+                                  IndexType n_cols,
+                                  const half* __restrict__ in,
+                                  half* __restrict__ out)
+{
+  __shared__ half tile[TILE_DIM][TILE_DIM + 1];
```

Comment:
Transpose of a non-symmetric non-square matrix is non-trivial to parallelize. This seems to be doing it all in a single kernel. This makes me wonder if we might be lacking test coverage here.